src/transformState.js

```diff
@@ -6,6 +6,21 @@
  */
 function transformState(state, actions) {
   // write code here
+  for (const i of actions) {
+    if (i.type === 'addProperties') {
+      Object.assign(state, ...[i.extraData]);
+    } else if (i.type === 'removeProperties') {
+      for (const j of i.keysToRemove) {
+        delete state[j];
+      }
+    } else if (i.type === 'clear') {
+      for (const key in state) {
+        delete state[key];
+      }
+    }
+  }
+
```

Comment:
better to use `switch` instead of if`s

Reply:
Using `switch` instead of `if` statements can be preferable in certain situations for a few reasons:

1. Readability: When you have multiple conditions to check, `switch` statements can make your code more readable by grouping similar cases together.

2. Efficiency: `switch` statements can often be more efficient than `if` statements when you have a large number of cases to check. This is because `switch` statements use jump tables or hash tables for lookups, which can be faster than checking each condition in turn.

3. Default case handling: `switch` statements have a built-in mechanism for handling cases that don't match any of the other conditions. This can be useful in situations where you need to handle unexpected inputs.

However, `if` statements still have their place and may be preferable in some cases, such as when you need to evaluate more complex conditions or when you want to perform additional operations before or after the condition is evaluated. Ultimately, the choice between `switch` and `if` statements depends on the specific requirements of your code.